### ISSUE TITLE
ADE: Add support for Generation 2 VMs

### ISSUE THREAD

**Issue by johanburati** [Azure/azure-linux-extensions#1212]
We currently cannot encrypt Gen 2 VMs:
```
ERROR: Azure Disk Encryption is not supported with Generation 2 Hypervisor.
```
Could you give an estimate when this will be supported ?

**Comment by jackdelab** [comment]
Disk encryption seems to be working with the az CLI for Generation 2 Hypervisor VMs, but not with the latest version of the SDK for Go (version 2021-07-01)

**Comment by k-byun** [comment]
I don't agree with @jackdelab , or maybe I am not using the correct cmdlet. I'm using `az vm encryption enable` but it's generating: 
> (NotSupported) Azure Disk Encryption is not supported with Generation 2 Hypervisor.
Code: NotSupported
Message: Azure Disk Encryption is not supported with Generation 2 Hypervisor.

**Comment by lbouriez** [comment]
I am having the same issue despite the documentation saying it should work. Did you find a workaround ?
